### PR TITLE
Support line numbers and github like line highlighted

### PR DESCRIPTION
**Pull request by rjoydip**
This PR addressed #210, #398 and #617. Implementation is followed GitHub like line references.

//cc @lucacasonato

### CODE REVIEW

File: components/CodeBlock.tsx [hunk 1]
```diff
@@ -25,7 +28,36 @@ export const RawCodeBlock = ({
...
+  useEffect(() => {
+    Router.events.on("hashChangeComplete", (url: any) => {
+      setHashValue(url.slice(url.indexOf("#")));
+    });
+    const { hash } = location;
+    setHashValue(hash);
+    return () => {
+      Router.events.off("hashChangeComplete", () => {});
+    };
+  }, []);
```
Comment (lucacasonato) [inline components/CodeBlock.tsx:43]: Can't you use the `useRouter` hook for this? Might not be possible, but seems cleaner.

Comment (rjoydip) [reply components/CodeBlock.tsx:43]: Okay sure

Comment (lucacasonato) [reply components/CodeBlock.tsx:43]: Oh yeah you're right. Looks good then :-)

**Comment by lucacasonato** [comment]
Can you merge with master to make CI happy?

**Comment by rjoydip** [comment]
//cc @leonardoelias Merged with master

**Pull request merged**